lxd/cluster/gateway.go

```diff
@@ -284,24 +285,14 @@ func (g *Gateway) HandlerFuncs(heartbeatHandler HeartbeatHandler, trustedCerts f
 			return
 		}
 
-		hijacker, ok := w.(http.Hijacker)
-		if !ok {
-			http.Error(w, "Webserver doesn't support hijacking", http.StatusInternalServerError)
-			return
-		}
-
-		conn, _, err := hijacker.Hijack()
+		conn, err := response.Upgrade(w, "dqlite")
 		if err != nil {
-			message := fmt.Errorf("Failed to hijack connection: %w", err).Error()
-			http.Error(w, message, http.StatusInternalServerError)
-			return
-		}
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+
+			if conn != nil {
+				conn.Close()
+			}
```

Comment:
In `sftp.go` you do
```
	conn, err := response.Upgrade(w, "sftp")
	if conn != nil {
		defer conn.Close()
	}
```

Shouldn't this also be outside of the error check block?

Reply:
It's not an error check block, it's a `conn != nil` block.

Reply:
The `conn != nil` check is inside of the `err != nil` block, according to the diff.

Reply:
I've fixed this now as conn should only be closed on error  and if returned